src/lean_spec/subspecs/containers/state/state.py

```diff
@@ -516,3 +517,53 @@ def state_transition(self, block: Block, valid_signatures: bool = True) -> "Stat
             raise AssertionError("Invalid block state root")
 
         return new_state
+
+    def build_block(
+        self,
+        slot: Slot,
+        proposer_index: ValidatorIndex,
+        parent_root: Bytes32,
+        attestations: list[Attestation] | None = None,
+    ) -> tuple[Block, "State"]:
+        """
+        Build a valid block on top of this state.
+
+        This is a pure function that:
+        1. Advances the state to the target slot
```

Comment:
Isn't this function kind of reverting this PR which was just merged https://github.com/leanEthereum/leanSpec/pull/212

In that we are doing redundant `process_slots(slot).process_block` again? In the `produce_block_with_signatures()` function